SConstruct

```diff
@@ -790,18 +850,26 @@ def add_unseeded_analysis(w):
     def cluster(c):
         part = c['partition']
         cp = part['cp']
-        return [{'id': 'clust-' + str(i),
-                 'sorted_index': i,
-                 'unique_ids': clust,
-                 'size': len(clust)}
-                for i, clust
-                # Sort by len (dec) and apply index i
-                in enumerate(sorted(c['partition']['clusters'], key=len, reverse=True))
-                # Select top N or any matching seeds of interest
-                if (len(clust) > 5) and (i < options['depth'] or has_seeds(clust, c)) and valid_cluster(cp, part, clust)]
-
-
-    # Finally call out to the separate cluster analyses as defined above
+        clusters = []
+        # Sort by len (dec) and apply index i
+        for i, clust in enumerate(sorted(part['clusters'], key=len, reverse=True)):
+            # Select top N or any matching seeds of interest
+            if (len(clust) > 5) and (i < options['depth'] or has_seeds(clust, c)):
+                annotation_list, _ = read_partition_file(part, c)
```

Comment:
The reason we were doing the whole `cp = part['cp']` business above is that we already ran `read_partition_file` in a step further up, and don't want to read it in again if we don't have to. Suggest doing something similar here with the annotation_list, as again, this stuff in the SConstruct process can't be amortized away through parallelization or whatnot.

Reply:
@metasoarous Makes sense. This is when I asked about the SConstruct process and how important it was in terms of optimizing which it turns out the answer as I understand it is "very, because it can be a bottleneck for large builds". But I believe I tested both and surprisingly, it turned out that doing it this way was faster. I suspect this is because if we want to use the fact that
> we already ran `read_partition_file`

 , that means (the way I conceived of it) we have to attach `annotation_list` to each partition object which is a large list of dicts that maybe slows down the processing of the partition. Though I could be mistaken in thinking that  
> we have to attach `annotation_list` to each partition object

Reply:
Yeah... the problem is probably that it's trying to then write it out in the metadata. If you hunt around (grep for cp), you'll notice that there's a line in there somewhere to make sure that cp gets left out of the metadata. If you do the same for `annotation_list` or whatever you call it, that should minimize the difference. It's possible that it's not as a big of a difference as I suspect, but it would be nice to verify that with a quick test.

Let me know if you have trouble finding that exclusion syntax, and I'll hunt it down for you in a bit.

Reply:
Ah ok. I will test that and let you know if I have questions about the details of excluding part of the partition metadata.

Reply:
@metasoarous it seems like it's pretty much the same (was actually very slightly slower) with passing the annotations through but "eliding" them here https://github.com/matsengrp/cft/blob/master/SConstruct#L769. I can push to this branch if you want to take a look because maybe I am not doing it the way you had in mind. This is on a dataset I believe has only unseeded clusters, which is I think where the optimization would apply.

Reply:
The same as which? The same as when you just read the partition file in a second time, or the same as prior to eliding the cached results of `read_partition_file`? Either way, I guess I'm a little surprised and would kind of like to take a look at the code.

Reply:
Hmm... maybe its ending up doing more work garbage collecting or something...

Reply:
Sorry, I meant the "same" as the code that is pushed here where 
> you just read the partition file in a second time

Will push the "eliding" version so you can take a look.